CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contribution Guide
+
+## Getting Started
+
+### Prerequisites
+Please make sure you have the following programs installed:
+- [Python 3.9](https://www.python.org/downloads/)
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Virtual Box](https://www.virtualbox.org/wiki/Downloads)
+- [Vagrant](https://www.vagrantup.com/downloads)
+
+### Setting up
+1. Clone your fork.
+2. Create a new branch.
+3. Open any terminal and navigate to the project directory.
+4. Run the `vagrant up` command.
+
+After the dependencies were installed and the VM initialized, you can proceed working on the project code.
+
+## Development Process
+
+### Commits
+- Separate subject from body with a blank line.
+- Limit the subject line to 50 characters.
+- Capitalize the subject line.
+- Do not end subject line with a period.
+- Use the imperative mood in the subject line.
+- Wrap the body at 72 characters.
+- Use the body to explain what and why vs. how.
+
+For more information and examples, please refer to [this guide](https://cbea.ms/git-commit/#seven-rules).
+
+### Issues
+- The title must be clear, and describe what the issue is about.
+- The description of the issue must include a clear explanation about the reasons for doing the work.
+- Status Labels(Optional):
+  - Status/New - An issue that was just created, in addition unlabeled issues are considered new.
+  - Status/Backlog - A issue that was discussed and accepted for work.
+  - Status/In progress - An issue that is being worked on.
+
+### Pull Requests
+- **The Pull Request should not break any of the existing functionality**
+- <ins>Description-</ins> the PR description should explain what changes have been made, and why. Incase it's related to a specific `issue` - it is mandatory to include a link to it.
+Please refer to this [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) of how to link a PR to an issue.
+- Each PR should include no more than 5 `commits`
+- The pull request will not undergo review until all tests have passed:
+    1. A [flake8](https://www.flake8rules.com/) workflow to make sure the PR conforms to proper Python syntax and coding standards.
+    2. Automatically checked by a CI system.
+- Each pull request requires the approval of at least <ins>2 team members</ins> before merging.
+- Each PR should [request reviewers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review). Reviewers should be people who are knowledgeable about the code you're modifying or related to the specific issue.
+Please refer to this [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) in order to assign users to PRs and issues.
```

Comment:
I'd break those long lines (here and on other places) to make it easier to read. You can arbitrarily break lines in markdown files and it won't create a newline on the rendered output (as long as you don't leave an empty line).

Basically, if you have links, then you won't break the link in the middle (but you can leave long links on lines of their own), but for all other content, you can break it at a common number that fits in an editor (89 characters is a nice number that typically fits having 2 documents side-by-side on the same monitor with reasonable text size - but it's up to you).


I'd suggest breaking 
```suggestion
- <ins>Description-</ins> the PR description should explain what changes have been made, and why.
In case it's related to a specific `issue` - it is mandatory to include a link to it. Please refer to this
 [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
 for how to link a PR to an issue.
- Each PR should include no more than 5 `commits`
- The pull request will not undergo review until all tests have passed:
    1. A [flake8](https://www.flake8rules.com/) workflow to make sure the PR conforms to proper Python syntax
        and coding standards.
    2. Automatically checked by a CI system.
- Each pull request requires the approval of at least <ins>2 team members</ins> before merging.
- Each PR should
   [request reviewers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review).
   Reviewers should be people who are knowledgeable about the code you're modifying or related to the
   specific issue.
Please refer to this
[guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users)
 in order to assign users to PRs and issues.
```